mcpgateway/plugins/framework/base.py

```diff
@@ -253,6 +255,38 @@ async def resource_post_fetch(self, payload: ResourcePostFetchPayload, context:
                                    """
         )
 
+    async def http_pre_forwarding_call(self, payload: HttpHeaderPayload, context: PluginContext) -> HttpHeaderPayloadResult:
+        """Plugin hook run before a forwarding HTTP call is made to a tool or gateway.
+
+        Args:
+            payload: The HTTP headers that can be checked or modified  .
+            context: Contextual information about the hook call.
+
+        Raises:
+            NotImplementedError: needs to be implemented by sub class.
+        """
+        raise NotImplementedError(
```

Comment:
This currently raises NotImplementedError. Consider whether a default "no-op" implementation (e.g., returning payload unchanged) might simplify plugin creation when HTTP hook logic isn’t needed.